src/web/lib/dailyArticleCount.test.ts

```diff
@@ -0,0 +1,153 @@
+import {
+    DailyArticleCount,
+    DailyArticleCountKey,
+    getDailyArticleCount,
+    incrementDailyArticleCount,
+} from './dailyArticleCount';
+
+const today = Math.floor(Date.now() / 86400000);
+
+const validDailyArticleCount: DailyArticleCount = [
+    {
+        day: today,
+        count: 3,
+    },
+    {
+        day: today - 1,
+        count: 2,
+    },
+    {
+        day: today - 2,
+        count: 1,
+    },
+];
+
+describe('dailyArticleCount', () => {
+    beforeEach(() => {
+        localStorage.clear();
+    });
+
+    it('gets an empty array of daily article counts by default', () => {
+        const output = getDailyArticleCount();
+
+        expect(output).toEqual([]);
+    });
+
+    it('returns array of valid daily article counts if they exist', () => {
+        localStorage.setItem(
+            DailyArticleCountKey,
+            JSON.stringify({ value: validDailyArticleCount }),
+        );
+
+        const output = getDailyArticleCount();
+
+        expect(output).toEqual(validDailyArticleCount);
+    });
+
+    it('returns empty array of daily article counts if failed to parse local storage, and remove the key from localStorage', () => {
+        localStorage.setItem(DailyArticleCountKey, 'not a valid json string');
+
+        const output = getDailyArticleCount();
+
+        expect(output).toEqual([]);
+        expect(localStorage.getItem(DailyArticleCountKey)).toBeNull();
+    });
+
+    it('returns empty array of daily article counts if invalid json format, and removes the key from localStorage', () => {
```

Comment:
this test now checks that the localStorage string is in the correct format when parsed